src/Microsoft.AspNet.Mvc.Core/Rendering/Expressions/CachedExpressionCompiler.cs

```diff
@@ -79,42 +82,43 @@ private static class Compiler<TIn, TOut>
                 // by around one microsecond, so it's not worth it to complicate the logic here with
                 // an architecture check.
 
-                var memberExpr = expr.Body as MemberExpression;
-                if (memberExpr != null)
+                var memberExpression = expression.Body as MemberExpression;
+                if (memberExpression != null)
                 {
-                    if (memberExpr.Expression == expr.Parameters[0] || memberExpr.Expression == null)
+                    if (memberExpression.Expression == expression.Parameters[0] || memberExpression.Expression == null)
                     {
                         // model => model.Member or model => StaticMember
-                        return _simpleMemberAccessDict.GetOrAdd(memberExpr.Member, _ => expr.Compile());
+                        return _simpleMemberAccessDict.GetOrAdd(memberExpression.Member, _ => expression.Compile());
                     }
 
-                    var constExpr = memberExpr.Expression as ConstantExpression;
-                    if (constExpr != null)
+                    var constantExpression = memberExpression.Expression as ConstantExpression;
+                    if (constantExpression != null)
                     {
                         // model => {const}.Member (captured local variable)
-                        var del = _constMemberAccessDict.GetOrAdd(memberExpr.Member, _ =>
+                        var compiledExpression = _constMemberAccessDict.GetOrAdd(memberExpression.Member, _ =>
                         {
                             // rewrite as capturedLocal => ((TDeclaringType)capturedLocal).Member
-                            var constParamExpr = Expression.Parameter(typeof(object), "capturedLocal");
-                            var constCastExpr = Expression.Convert(constParamExpr, memberExpr.Member.DeclaringType);
-                            var newMemberAccessExpr = memberExpr.Update(constCastExpr);
-                            var newLambdaExpr =
-                                Expression.Lambda<Func<object, TOut>>(newMemberAccessExpr, constParamExpr);
-                            return newLambdaExpr.Compile();
+                            var parameterExpression = Expression.Parameter(typeof(object), "capturedLocal");
+                            var castExpression =
+                                Expression.Convert(parameterExpression, memberExpression.Member.DeclaringType);
+                            var newMemberExpression = memberExpression.Update(castExpression);
+                            var newExpression =
```

Comment:
never seen `new` used in this format and can make a reader look twice: aka

`var foo = new MyClass();` vs
`var foo = newMyVariable`.